deeppavlov/pipeline_manager/observer.py

```diff
@@ -0,0 +1,229 @@
+from collections import OrderedDict
+from pathlib import Path
+from shutil import rmtree
+from typing import Union
+
+
+class Observer(object):
+    """
+    The :class:`~pipeline_manager.observer.Observer` implements the functions of observing the course of experiments,
+    collecting results, time and other useful information, logging and storing it.
+    """
+
+    def __init__(self, name: str, root: Union[str, Path], info: dict, date: str, plot: bool) -> None:
+        """
+        Initializes the log, creates a folders tree and files necessary for the observer to work.
+
+        Args:
+            name: str; name of the experiments.
+            root: str; path to root folder.
+            info: dict; additional information that you want to add to the log, the content of the dictionary
+             does not affect the algorithm
+            date: str; date of the experiment.
+        """
+
+        self.exp_name = name
+        self.exp_inf = info
+        self.root = root
+        self.date = date
+        self.plot = plot
+
+        # tmp parameters
+        self.pipe_ind = 0
+        self.pipe_conf = None
+        self.model = None
+        self.pipe_res = None
+        self.pipe_time = None
+        self.batch_size = None
+        self.dataset = None
+
+        # build folder dependencies
+        self.save_path = self.root / self.date / self.exp_name / 'checkpoints'
+        self.log_path = self.root / date / self.exp_name
+        self.log_file = self.log_path / (self.exp_name + '.json')
+
+        if not self.save_path.is_dir():
+            self.save_path.mkdir(parents=True)
+        if self.plot:
+            if not (self.log_path / 'images').is_dir():
+                (self.log_path / 'images').mkdir()
+
+        self.log = OrderedDict(experiment_info=OrderedDict(date=date,
+                                                           exp_name=self.exp_name,
+                                                           root=str(self.root),
+                                                           info=self.exp_inf,
+                                                           number_of_pipes=None,
+                                                           metrics=None,
+                                                           target_metric=None),
+                               experiments=OrderedDict())
+
+    def tmp_reset(self) -> None:
+        """
+        Reinitialize temporary attributes.
+        """
+        # tmp parameters
+        self.pipe_ind = 0
+        self.pipe_conf = None
+        self.model = None
+        self.pipe_res = None
+        self.pipe_time = None
+        self.batch_size = None
+        self.dataset = None
+
+    def write(self) -> None:
+        """
+        Write log in log_file.
+        """
+        if self.log_file.is_file():
+            with open(str(self.log_file), 'r') as old_log:
+                old_log = json.load(old_log)
+
+            new_log = self.merge_logs(old_log, self.log)
```

Comment:
If new logs depend on old ones, you should use at least sqlite instead of json.
If logs are independent, switch to jsonlines

Reply:
Switch to jsonlines.